src/pipdeptree/_discovery.py

```diff
@@ -1,26 +1,37 @@
 from __future__ import annotations
 
+import ast
 import site
+import subprocess  # noqa: S404
 import sys
 from importlib.metadata import Distribution, distributions
+from pathlib import Path
 from typing import Iterable, Tuple
 
 from packaging.utils import canonicalize_name
 
 
 def get_installed_distributions(
+    interpreter: str = str(sys.executable),
     local_only: bool = False,  # noqa: FBT001, FBT002
     user_only: bool = False,  # noqa: FBT001, FBT002
 ) -> list[Distribution]:
     # See https://docs.python.org/3/library/venv.html#how-venvs-work for more details.
     in_venv = sys.prefix != sys.base_prefix
     original_dists: Iterable[Distribution] = []
+    py_path = Path(interpreter).absolute()
 
     if local_only and in_venv:
         venv_site_packages = site.getsitepackages([sys.prefix])
         original_dists = distributions(path=venv_site_packages)
     elif user_only:
         original_dists = distributions(path=[site.getusersitepackages()])
+    elif py_path != Path(sys.executable).absolute():
```

Comment:
Building on top of my previous comment, I think we could do something similar to the following:
```suggestion
   using_custom_interpreter = py_path != Path(sys.executable).absolute()
   if using_custom_interpreter:
       args = []
       if local_only:
           args = ["str(py_path), "-c", "import sys, site; print(site.getsitepackages([sys.prefix]))"]
       else:
           cmd = [str(py_path), "-c", "import sys, site; print(site.getsitepackages([sys.prefix]))"]
        # Add the custom python subprocess run code here
    if local_only and in_venv:
        venv_site_packages = site.getsitepackages([sys.prefix])
        original_dists = distributions(path=venv_site_packages)
    elif user_only:
        original_dists = distributions(path=[site.getusersitepackages()])
```